src/Fmt_ast.ml

```diff
@@ -1713,16 +1713,34 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext ({ast= exp} as xexp)
         $ fmt_atrs )
   | Pexp_open (flag, {txt; loc}, e0) ->
       let override = Poly.(flag = Override) in
+      let let_open =
+        match c.conf.let_open with
+        | `Preserve ->
+            let filter = function Parser.IN -> true | _ -> false in
+            let from = loc and upto = e0.pexp_loc in
+            let toks =
+              if
+                Source_code_position.ascending from.loc_start upto.loc_start
+                < 0
+              then Source.tokens_between c.source ~from ~upto ~filter
+              else []
+            in
+            if List.is_empty toks then `Short else `Long
```

Comment:
It would be nice to outline this token checking into a function named something like `source_contains_IN_token_between ~from ~upto`